src/aiidalab_qe/setup/codes.py

```diff
@@ -44,18 +45,45 @@ def get_qe_env():
 
 
 def qe_installed():
-    import json
+    """Check if Quantum Espresso (QE) is installed in the specified conda environment.
 
-    env_exist = get_qe_env().exists()
-    proc = subprocess.run(
-        ["conda", "list", "-n", f"{get_qe_env().name}", "--json", "--full-name", "qe"],
-        check=True,
-        capture_output=True,
-    )
+    Returns:
+        bool: True if the environment exists and QE is installed; False otherwise.
+    """
+    try:
+        # Verify if the specified conda environment exists
+        env_exist = get_qe_env().exists()
+
+        if not env_exist:
+            return False
+
+        # Run the conda list command to check for the QE package
+        proc = subprocess.run(
+            [
+                "conda",
+                "list",
+                "-n",
+                f"{get_qe_env().name}",
+                "--json",
+                "--full-name",
+                "qe",
+            ],
+            check=True,
+            capture_output=True,
+        )
 
-    info = json.loads(str(proc.stdout.decode()))[0]
+        # Load and interpret the JSON output
+        info = json.loads(proc.stdout.decode())
 
-    return env_exist and "qe" == info["name"]
+        # Check if 'qe' is listed in the environment
+        for package in info:
+            if package.get("name") == "qe":
+                return True
+        return False  # noqa: TRY300
+    except Exception as error:
+        raise RuntimeError(
+            "Failed to check if Quantum Espresso is installed."
+        ) from error
```

Comment:
```suggestion
    except Exception as exc:
        raise RuntimeError(
            "Failed to check if Quantum Espresso is installed."
        ) from exc
   else:
       return False  # noqa: TRY300
```

Also for the catched exception, I think there are two types of exceptions we know which are `SubprocessCallError` and `JsonParseError`. You can catch those two and for uncatched exceptions, use raise to propagated it up.

Reply:
For TRY300, you can move it to else block, https://docs.astral.sh/ruff/rules/try-consider-else/